sdk/storage/Azure.Storage.Common/src/Shared/StorageContext.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core.Pipeline;
+
+namespace Azure.Storage.Shared
+{
+    /// <summary>
+    /// Parent class of the *StorageContext classes.
+    /// Contains common properties used to create clients.
+    /// </summary>
+    internal class StorageContext
```

Comment:
I like the direction this PR is going. However the `Context` sounds like an invitation for "hey dump anything here".
Should we call it `StorageClientState`/`StorageClientOptionsInternal`/`StorageClientConfiguration`/`StorageClientProperties` ?